snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedDescriptionAcceptabilityUpdateRequest.java

```diff
@@ -59,49 +63,95 @@ protected boolean canUpdate(TransactionContext context) {
+					.collect(Collectors.toList());
 			
-			final SnomedRefSetMemberIndexEntry.Builder updatedMember = SnomedRefSetMemberIndexEntry.builder(existingMember);
-			final SnomedRefSetMemberIndexEntry oldRevision = updatedMember.build();
+			SnomedReferenceSetMember existingMember = null;
 			
-			if (acceptability.equals(newLanguageMembersToCreate.get(languageReferenceSetId))) {
-				// Exact match: make sure that the member is active
-				if (ensureMemberActive(context, existingMember, updatedMember)) {
+			if (!refsetMatches.isEmpty()) {
+				// Case 1: Match(es) exist for language reference set
+				final List<SnomedReferenceSetMember> acceptabilityMatches = refsetMatches.stream()
+						.filter(m -> Objects.equal(newAcceptability.getConceptId(), m.getProperties().get(SnomedRf2Headers.FIELD_ACCEPTABILITY_ID)))
+						.collect(Collectors.toList());
+				
+				if (acceptabilityMatches.isEmpty()) {
+					// Case 1.1: No match for acceptability; use an existing member preferably an extension one
+					existingMember = select(refsetMatches, moduleId, Concepts.MODULE_SCT_CORE);
+					final SnomedRefSetMemberIndexEntry.Builder updatedMember = SnomedRefSetMemberIndexEntry.builder(existingMember);
+					final SnomedRefSetMemberIndexEntry oldRevision = updatedMember.build();
+					
+					// Change acceptability, set status to active if required, place it in the supplied module
+					updatedMember.field(SnomedRf2Headers.FIELD_ACCEPTABILITY_ID, newAcceptability.getConceptId());
+					ensureMemberActive(context, existingMember, updatedMember);
+					updateModule(context, existingMember, updatedMember, moduleIdSupplier.apply(componentToUpdate));
+					unsetEffectiveTime(existingMember, updatedMember);
 					context.update(oldRevision, updatedMember.build());
-				}
 
-				// Remove it from the working list, as we have found a match
-				newLanguageMembersToCreate.remove(languageReferenceSetId);
-			} else if (newLanguageMembersToCreate.containsKey(languageReferenceSetId)) {
-				// Change acceptability, set status to active if required, place it in the supplied module
-				final Acceptability newAcceptability = newLanguageMembersToCreate.get(languageReferenceSetId);
-				updatedMember.field(SnomedRf2Headers.FIELD_ACCEPTABILITY_ID, newAcceptability.getConceptId());
-				ensureMemberActive(context, existingMember, updatedMember);
-				updateModule(context, existingMember, updatedMember, moduleIdSupplier.apply(componentToUpdate));
-				unsetEffectiveTime(existingMember, updatedMember);
-				context.update(oldRevision, updatedMember.build());
-				newLanguageMembersToCreate.remove(languageReferenceSetId);
+				} else {
+					// Case 1.2: There is at least one match for acceptability
+					final List<SnomedReferenceSetMember> activeMatches = acceptabilityMatches.stream()
+							.filter(m -> Objects.equal(newAcceptability.getConceptId(), m.getProperties().get(SnomedRf2Headers.FIELD_ACCEPTABILITY_ID)))
```

Comment:
This filter should look for active members.

Reply:
SnomedDescriptionApiTest no longer fails after this fix.